foreign_cc/private/configure_vars.bzl

```diff
@@ -0,0 +1,137 @@
+"""Helper methods to assembler configure env variables from Bazel information."""
```

Comment:
Is the issue in `make` or `configure_make`? I think that between the two rules `configure_make` should generally be an extension of `make` and should try to call functionality from that rule as much as possible. Is all the configuration done here at all specific to `configure_make`?

Reply:
The issue is in `make`: `configure_make` correctly sets the usual env variables before invoking the `configure` script (which likely uses these variables to generate the Makefile). `make`, without this PR, never sets any variables. I tried to extract and reuse the common functionality.

Reply:
Could the semantics of the file and symbols within it be changed from `configure_make` to `make`? I'm most concerned in understanding what is required to get `make` specifically to work and what things are specific to `configure` or any of the tooling there.

Reply:
We could rename everything here to `env_vars` rather than `configure_vars`. Nothing seems specific to configure, it's just that the variables passed to `make` are the same that are typically used to influence what a `configure` script does. 

Would you want me to carry out the renaming?

Reply:
But otherwise, yeah. I think the rename would be good

Reply:
I carried out the rename in https://github.com/bazelbuild/rules_foreign_cc/pull/777/commits/ef8d5b13837c0cce36fd93d627dc73cad74394a3. Could you take a look?

Reply:
Fantastic! Thank you!